addons/web_editor/static/src/js/editor/snippets.editor.js

```diff
@@ -2947,7 +2947,18 @@ var SnippetsMenu = Widget.extend({
         if (!ev.currentTarget.classList.contains('active')) {
             this._activateSnippet(false);
             this._mutex.exec(() => {
-                this._activateEmptyOptionsTab();
+                let $snippet = $();
+                let i = 0;
+                const pageOptions = this.templateOptions.filter(template => template.data.pageOptions);
+                while (!$snippet.length && i < pageOptions.length) {
+                    $snippet = pageOptions[i].selector.all(this.$body);
```

Comment:
@bvr-odoo Is using `this.$body` necessary? Can't you just call `all` without param?

Reply:
Indeed it is not necessary because without param, `all` searches in the editable area.